File: src/ProtectionTools.WebUI/Controllers/AppController.cs
namespace ProtectionTools.WebUI.Controllers {
    using System.Collections.Generic;
    using System.Linq;
    using Core.Elements.Connections.Factory;
    using Core.Elements.ElectroReceivers;
    using Microsoft.AspNet.Mvc;
    using Services;
    using ViewModels;

    public class AppController : Controller {
        private readonly IBusService _busService;
        private readonly IConnectionFactory _connectionFactory;

        public AppController(IBusService busService, IConnectionFactory connectionFactory) {
            _busService = busService;
            _connectionFactory = connectionFactory;
        }

        public IActionResult Index() {
            return View();
        }

        public IActionResult Calc([FromBody] BusViewModel model) {
            if (!ModelState.IsValid) {
                ModelState.AddModelError("", "Please enter correct data");
                return View(model);
            }
            model.Amperage = _busService.GetCurrent(model.PowerCoef, model.NominalVoltage,
                model.Elements.Select(
                    elem =>
                        new ElectroReceiver(
                            _connectionFactory,
                            elem.ActivePower,
                            elem.Count,
                            elem.UsingCoefficient,
                            elem.Cos)));
            return Json(model);
        }

        public IActionResult Add([FromBody] List<ElementViewModel> model) {
            model.Add(new ElementViewModel());
            return Json(model);
        }
    }
}

File: src/ProtectionTools.WebUI/Scripts/controllers/appController.js
﻿(function() {
    var app = angular.module("electricityApp");
    var appController = function($scope, electricity) {
        $scope.message = "helloworld";
        $scope.model = {
            PowerCoef: 1.18,
            NominalVoltage: 0.38,
            Elements: []
        }

        $scope.calc = function () {
            electricity.calc($scope.model).then(function (data) {
                $scope.model = data;
            })
        }

        $scope.addElement = function() {
            electricity.addElement($scope.model)();
        }
    }
    app.controller("AppController", appController);
})();
